lib/matplotlib/axes/_axes.py

```diff
@@ -2420,22 +2423,38 @@ def stem(self, *args, **kwargs):
         except IndexError:
             basefmt = kwargs.pop('basefmt', 'r-')
 
+        # Check if the user wants a vertical stem plot
+        vertical = kwargs.pop('vertical', None)
+        if vertical is None:
+            vertical = False
+
```

Comment:
It is a bigger change, but could you refactor this to have on `if` block up here which does the flipping (and maybe rename x, y?) and then the rest of the logic is the same.